### PR TITLE
add s1meta.doppler_estimate (Doppler Centroid Estimate annotations)

### PR DESCRIPTION
**Pull request by agrouaze**
address issue #49 to add Doppler Centroid Estimate annotations in `s1meta`.
This work has already been done in #38 but this PR is meant to ease the validation/integration with official develop branch.

- [x] add compound var in `sentinel1_xml_mappings.pt`
- [x] add property `s1meta.doppler_estimate` in `sentinel1_meta.py`
- [x] validation of the new property

### CODE REVIEW

**Comment by agrouaze** [comment]
image

**Comment by oarcher** [comment]
From `support/s1-level-1-product.xsd` file, the documentation for `dcEstimate` is:

> Doppler centroid estimate record which contains the Doppler centroid calculated from geometry and estimated from the data, associated signal-to-noise ratio values and indicates which DCE method was used by the IPF during image processing. With a minimum Doppler centroid update rate of 1s (for IW and EW where the Doppler is recalculated for every burst cycle) and a maximum product length of 25 minutes, the maximum size of this list is 1500 elements.


So is `n_estimates` is a burst index ?

**Comment by agrouaze** [comment]
As it is said the Doppler Centroid is estimated at each burst cycle but for instance in GRD product you have no burst (`<burstList count="0"/>` ) and you have 27 doppler centroid estimates ( `<dcEstimateList count="27">`)
So `n_estimates` is almost a burst index but I think it has to be separated from bursts informations.
Note: even for SLC products you can find 9 bursts for 11 DCE.

**Comment by oarcher** [comment]
`geo_polynom` and `data_polynom` seems to be polynomial coefficients. Do you think that they can be replaced by a [numpy.polynomial](https://numpy.org/doc/stable/reference/routines.polynomials.html), as done in https://github.com/agrouaze/xsar/blob/a632ea770652b1feb93d82a4ed916bb2e355f073/src/xsar/sentinel1_xml_mappings.py#L487 ?

If so, it might be possible to remove those dimensions.


So the only dimensions that require this structure to be a xarray instead of a pandas dataframe is `nb_fine_dce`, used for 2 variables (`slant_range_time` and `frequency`)

**Comment by agrouaze** [comment]
Yes `numpy.polynomial` is may be preferable I do the change.

**Comment by agrouaze** [comment]
image

**Review by oarcher — CHANGES_REQUESTED** [review]

File: src/xsar/sentinel1_xml_mappings.py [hunk 1]
```diff
@@ -31,6 +32,7 @@
 datetime64_array = lambda x: np.array([np.datetime64(date_converter([sx])) for sx in x])
 int_1Darray_from_string = lambda x: np.fromstring(x[0], dtype=int, sep=' ')
 float_2Darray_from_string_list = lambda x: np.vstack([np.fromstring(e, dtype=float, sep=' ') for e in x])
+float_list_of_list_from_string = lambda x: [np.fromstring(e, dtype=float, sep=' ') for e in x]
```
[the remark this answers is not in the document]
Comment (agrouaze) [reply src/xsar/sentinel1_xml_mappings.py]: accepted & fixed.

**Comment by agrouaze** [comment]
yes ready for merge!

**Pull request merged**